src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -53,11 +77,29 @@
             Ref("DoubleQuotedLiteralSegment"),
         ]
     ),
-    PostTableExpressionGrammar=Sequence(
-        OneOf("IGNORE", "FORCE", "USE"),
-        OneOf("INDEX", "KEY"),
-        Sequence("FOR", OneOf("JOIN"), optional=True),
-        Bracketed(Ref("ObjectReferenceSegment")),
+    FromClauseTerminatorGrammar=OneOf(
+        "WHERE",
+        "LIMIT",
+        "GROUP",
+        "ORDER",
+        "HAVING",
+        "QUALIFY",
+        "WINDOW",
+        Ref("IndexHintClauseSegment"),
```

Comment:
It looks like this element is the same as the ANSI dialect except for lines 88-90. Should we look at using `.copy()` to base this segment on the ANSI one, to reduce copy/paste? (NOTE: I _think_ this is supported for grammars as well as segments.)

Reply:
Will do, good suggestion.